osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs

```diff
@@ -60,11 +60,23 @@ public override void Add(DrawableHitObject h)
 
             var c = h as IDrawableHitObjectWithProxiedApproach;
             if (c != null)
-                approachCircles.Add(c.ProxiedLayer.CreateProxy());
+            {
+                var original = c.ProxiedLayer;
+                // lifetime is set on LoadComplete so wait until it.
+                original.OnLoadComplete += addApproachCircleProxy;
+            }
 
             base.Add(h);
         }
 
+        private void addApproachCircleProxy(Drawable d)
+        {
+            var proxy = d.CreateProxy();
+            proxy.LifetimeStart = d.LifetimeStart;
+            proxy.LifetimeEnd = d.LifetimeEnd;
```

Comment:
This stuff shouldn't be required, I think? ProxyDrawable internally does:

`protected internal override bool ShouldBeAlive => base.ShouldBeAlive && Original.ShouldBeAlive;`

Reply:
This is required because `LifetimeManagementContainer` uses `LifetimeStart` and `LifetimeEnd` to determine lifetime and not `ShouldBeAlive`.

Reply:
Consider that the hitobjects are going to need to remain alive while they haven't been hit (or while they haven't been fully rewinded). I was planning on overriding `ShouldBeAlive` to achieve this.

Reply:
Although arbitrary override can't be supported, it is possible to support dynamically changing lifetime in `LifetimeManagementContainer`, including making thing dead at any time. But that complicates the implementation. Thus I leaved the restriction for now.